nservicebus/upgrades/5to6/distributor.md

```diff
@@ -0,0 +1,128 @@
+
+WARNING: Following this process when endpoints subscribe to events may cause duplicate events or message loss!
+
+* Upgrade all endpoints that interact with the distributor to Version 6 first. At this stage, do not upgrade the workers to Version 6.
+  * upgrade all endpoints that send command messages to the Distributor endpoint to be distributed,
+  * upgrade all endpoints that send subscription messages to the Distributor, 
+  * upgrade all endpoints subscribing to events published by worker nodes.
+* Configure all mentioned endpoints above to use [sender-side distribution](/nservicebus/msmq/scalability-and-ha/sender-side-distribution.md) to route messages directly to the workers instead of the Distributor.
+* Ensure no more messages are routed to the Distributor.
+* Apply the following steps for each worker:
+  * Shut down the worker.
+  * [Upgrade to NServiceBus Version 6](#upgrade-endpoint-to-version-6).
+  * Start it again.
+* Shut down and remove the Distributor.
+
+
+## Upgrade endpoint to Version 6
+
+* Remove the `NServiceBus.Distributor.MSMQ` package.
+* Remove Distributor specific configuration options.
+  * Remove the `MasterNodeConfig` section from the application configuration file.
+  * Remove the `DistributorControlAddress` and `DistributorDataAddress` attributes from the `UnicastBusConfig` configuration section in the application configuration file.
+* Upgrade the endpoint to Version 6 (See the [Upgrade Guide](/nservicebus/upgrades/5to6).
+
+
+## Remove Distributor subscriptions
+
+When decommissioning the Distributor, it is necessary to manually remove remaining subscriptions in case the Distributor receives events to avoid issues on publishers or events being lost.
+
+Follow the steps specified by the used subscription storage:
+
+
+### Removing subscriptions from RavenDB persistence
+
+For every endpoint the Distributor (including its workers) subscribes to:
+* Find the publishers database in the RavenDB Management Studio.
+* Go to the Query view.
+* From the indexes select `Subscriptions`
+* Enter the following Query: `Subscribers:*"<endpointAddress>"*` where you replace `<endpointAddress>` with the address of your endpoint. Make sure to escape special characters like `.` or `-` with `\`. E.g. `my\.endpoint@distributor\-machine`.
+* Run the query.
+* On all listed documents, remove the entry with the Distributor's subscriber address and save it.
+
+
+### Removing subscriptions from NHibernate persistence
+
+Execute the following script against the database which is configured for NHibernate persistence:
+
+```
+DELETE
+FROM dbo.Subscription
+WHERE SubscriberEndpoint = '<distributorAddress>'
+```
```

Comment:
@Particular/nhibernate-persistence-maintainers is that a valid/correct approach?